### ISSUE TITLE
cl-bench: conversion to floats of ratios with numerator/denominator > largest float -> #<single-float quiet NaN>

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#595]
```lisp
COMMON-LISP-USER> (float (/ (expt 10 300)(1+ (expt 10 300))))
1.0
COMMON-LISP-USER> (float (/ (expt 10 400) (1+  (expt 10 400))))
#<single-float quiet NaN>
````
in ecl
```lisp
> (float (/ (expt 10 300)(1+ (expt 10 300))))
1.0
> (float (/ (expt 10 400) (1+  (expt 10 400))))
1.0
> most-positive-double-float
1.7976931348623157d308
````

**Comment by kpoeck** [comment]
Seems to ecl has a more intelligent routine to convert a ratio to a float
```c
static double
ratio_to_double(cl_object num, cl_object den)
{
  cl_fixnum scale;
  cl_object bits = prepare_ratio_to_float(num, den, DBL_MANT_DIG, &scale);
#if (FIXNUM_BITS-ECL_TAG_BITS) >= DBL_MANT_DIG
  /* The output of prepare_ratio_to_float will always fit an integer */
  double output = ecl_fixnum(bits);
#else
  double output = ECL_FIXNUMP(bits)? ecl_fixnum(bits) : _ecl_big_to_double(bits);
#endif
  return ldexp(output, scale);
}
````

**Comment by kpoeck** [comment]
solved in #997
